src/main/java/org/truffleruby/core/proc/ProcNodes.java

```diff
@@ -274,6 +276,37 @@ protected RubyProc createSameArityProc(RubyProc userProc, RubyProc block) {
         }
     }
 
+    @Primitive(name = "proc_specify_arity", lowerFixnum = 1)
+    public abstract static class ProcSpecifyArityNode extends PrimitiveArrayArgumentsNode {
+        @Specialization
+        protected RubyProc specifyArityProc(RubyProc block, int argc) {
+            Arity oldArity = block.sharedMethodInfo.getArity();
+            final Arity newArity;
+            if (argc <= -1) {
+                newArity = new Arity(0, oldArity.getOptional(), true);
```

Comment:
Yes we possibly need to add specific tests for `-1`, `-2`, `0`, and then whatever MRI does on `-3`? An error?

Reply:
The caller should pass -1 when rb_define_method(..., -2).
Then values <= -1 mean `-argc-1` required arguments + rest.
For example, `-> (a,*) {}.arity => -2`.

Reply:
It would be great to add specs for `-2, -1, 0, 2` for `rb_define_method`, I think right now there is only for `0`.
Only add for `rb_define_method`, not for other `rb_define_*` since anyway they all end up in the same logic.

Reply:
There was no specs added and now this regressed, whooops! I'll add specs.

Reply:
My mistake, sorry, there were specs added in a separate PR, https://github.com/oracle/truffleruby/pull/2279.
Somehow they didn't catch the regression (https://github.com/oracle/truffleruby/pull/2662#issuecomment-1136962184), I'll investigate.

Reply:
It's simpler than I thought, https://github.com/oracle/truffleruby/pull/2279/files#diff-d3c9afdbdb36c78af232fbfd12a23fa196a4549837eb8b61e5c0eaf4ba3e73b7 added specs but also tags, and the tags were not removed, hence the specs didn't run.